services/bankid-api/lambdas/sign.js

```diff
@@ -1,42 +1,59 @@
 import { failure, success } from '../../../libs/response';
 import params from '../../../libs/params';
+import config from '../../../config';
+import { throwError } from '@helsingborg-stad/npm-api-error-handling';
 import * as request from '../../../libs/request';
 import * as bankId from '../helpers/bankId';
+import to from 'await-to-js';
+import snakeCaseKeys from 'snakecase-keys';
 
-const SSMParams = params.read('/bankidEnvs/dev');
+const SSMParams = params.read(config.bankId.envsKeyName);
 
 export const main = async event => {
-  try {
-    const bankidSSMParams = await SSMParams;
-    const { endUserIp, personalNumber, userVisibleData } = JSON.parse(event.body);
-
-    const payload = {
-      endUserIp,
-      personalNumber,
-      userVisibleData: userVisibleData
-        ? Buffer.from(userVisibleData).toString('base64') // eslint-disable-line no-undef
-        : undefined,
-    };
+  const bankidSSMParams = await SSMParams;
+  const { endUserIp, personalNumber, userVisibleData } = JSON.parse(event.body);
 
-    const bankIdClient = await bankId.client(bankidSSMParams);
-
-    const { data } = await request.call(
-      bankIdClient,
-      'post',
-      bankId.url(bankidSSMParams.apiUrl, '/sign'),
-      payload
-    );
-
-    const { orderRef, autoStartToken } = data;
-
-    return success({
-      type: 'bankIdSign',
-      attributes: {
-        order_ref: orderRef,
-        auto_start_token: autoStartToken,
-      },
-    });
-  } catch (error) {
-    return failure({ status: false, error: error.message });
+  if (!endUserIp || !userVisibleData) {
+    const error = {
+      status: 400,
+      code: 400,
+      name: 'BadRequestError',
+      detail: 'The server cannot process the request',
+      message: 'Missing endUserIp or userVisibleData',
+    };
+    return failure(error);
```

Comment:
would be nice if we could make a more general validation utility function but that is out of scope for this PR

Reply:
Or maybe not? 
Could you try to write a function that takes a callback function where we pass the throwError function?

My thought is that it could be used like this:
` async validateEventBody(error, callbackFunction(event.body)) {
   ...
   if(!callbackFunction()){
       throwError(...error)
   }
  ....
}`